app/models/author.rb

```diff
@@ -150,12 +150,14 @@ def self.build_attribute_hash_from_cap_profile(auth_hash)
     seed_hash
   end
 
-  def self.fetch_from_cap_and_create(profile_id, cap_client = nil)
-    cap_client = Cap::Client.new unless cap_client.present?
+  # @param [String] CAP Profile ID
+  # @param [Cap::Client] cap_client
+  # @return [Author] newly fetched, created and saved Author object
+  def self.fetch_from_cap_and_create(profile_id, cap_client = Cap::Client.new)
     profile_hash = cap_client.get_auth_profile(profile_id)
     a = Author.new
     a.update_from_cap_authorship_profile_hash(profile_hash)
-    a.save! # TODO: does this nullify `new_record?` later on, and why save now?
+    a.save!
     a
   end
```

Comment:
For the yardoc - it could add the `@raise` info about AR exceptions that `save!` could generate.  (Assuming this method does not intend to handle any of those exceptions here.)

Reply:
This addition to yardoc might be considered superfluous, because this is an AR model and clients should generally expect it to raise AR exceptions.  I think the complication is that this is doing some HTTP calls that might generate some non-AR exceptions (depending on where they get handled).  Like a lot of "utility" methods in this app, this one may help to DRY the code, but might add confusing responsibility to the AR model; the responsibility for this method really belongs somewhere in a CAP harvester or something, I guess.  Anyhow, this change is not intended to handle that, but adding the yardoc prompts such thoughts about it.

Reply:
Yeah, I agree w/ your line of thought here.